### ISSUE TITLE
Installer issue v1.7.2 - Not showing up in 140>DTS

### ISSUE THREAD

**Issue by ibrar2017** [keif888/SSISMHash#9]
Hi Keith,

I have installed Multiplehash v1.7.2 as Complete install as previously had v1.7.0 which has been uninstalled.

The issue is the 1.7.2 is not showing up in: C:\Program Files (x86)\Microsoft SQL Server\140\DTS\PipelineComponents and
C:\Program Files (x86)\Microsoft SQL Server\140\DTS\UpgradeMappings are missing the xml extentions.

However it successfully shows up on all other folders e.g. 100,110,120,130.

Is their an issue with the installer? And how do we overcome this issue? 

As I expected this to appear in their and avoid any upgrade errors due to target server version defaulting to SQL Server 2017.

Thanks

**Comment by keif888** [comment]
I have been unable to replicate this issue.
On a clean Windows 10 x64 install, with SSDT 17.2 installed, a complete installation of 1.7.2 places the DLL and XML files into 110, 120, 130 and 140.  Each DLL has the version (2012, 2014, 2016, and 2017) as appropriate.
SSISMHash.xml is installed into the 110, 120, 130 and 140 folders.  SSISMHash.extensions.xml is installed into the 120, 130 and 140 folders.

**Comment by keif888** [comment]
I have been able to replicate this issue for SSDT 15.5.2, using Visual Studio 2017, on a clean install without SSDT 17.2 installed.

This is caused by SSDT 15.5.2 as it is NOT installing the SQL 2017 DLL's or Registry keys, unlike SSDT 17.2.

Using a custom installation and enabling SQL 2017 will install the DLL's and allow this to work with SSDT 15.5.2 and Visual Studio 2017.  (I have tested this workaround)

I don't currently have a reliable fix for this as I am using the registry to detect which versions to target.

I have raised an question on the SQL Server setup and upgrade list.
https://social.msdn.microsoft.com/Forums/en-US/4822357f-b5b6-4923-8971-7a0ac9c07dc5/ssdt-1552-does-not-register-integrationservicespackage140-or-install-any-sql-2017-files?forum=sqlsetupandupgrade
